contracts/contracts/strategies/sonic/SonicStakingStrategy.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SonicValidatorDelegator } from "./SonicValidatorDelegator.sol";
+import { IWrappedSonic } from "../../interfaces/sonic/IWrappedSonic.sol";
+
+/**
+ * @title Staking Strategy for Sonic's native S currency
+ * @author Origin Protocol Inc
+ */
+contract SonicStakingStrategy is SonicValidatorDelegator {
+    // For future use
+    uint256[50] private __gap;
+
+    constructor(
+        BaseStrategyConfig memory _baseConfig,
+        address _wrappedSonic,
+        address _sfc
+    ) SonicValidatorDelegator(_baseConfig, _wrappedSonic, _sfc) {}
+
+    /// @notice Unlike other strategies, this does not deposit assets into the underlying platform.
+    /// It just checks the asset is Wrapped Sonic (wS) and emits the Deposit event.
+    /// To deposit wS into validators `delegate` must be used.
+    /// @param _asset Address of asset to deposit. Has to be Wrapped Sonic (wS).
+    /// @param _amount Amount of assets that were transferred to the strategy by the vault.
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        require(_asset == wrappedSonic, "Unsupported asset");
+        _deposit(_asset, _amount);
+    }
+
+    /**
+     * @notice Deposit Wrapped Sonic (wS) to this strategy so it can later be delegated to a validator.
+     * @param _asset Address of Wrapped Sonic (wS) token
+     * @param _amount Amount of Wrapped Sonic (wS) to deposit
+     */
+    function _deposit(address _asset, uint256 _amount) internal virtual {
+        require(_amount > 0, "Must deposit something");
+
+        _delegate(_amount);
+        emit Deposit(_asset, address(0), _amount);
+    }
+
+    /**
+     * @notice Deposit the entire balance of wrapped S in this strategy contract
+     */
+    /// @notice Deposits the WS asset into the underlying platform
+    function depositAll() external virtual override onlyVault nonReentrant {
+        uint256 wSBalance = IERC20(wrappedSonic).balanceOf(address(this));
+
+        if (wSBalance > 0) {
+            _deposit(wrappedSonic, wSBalance);
+        }
+    }
+
+    /// @notice Withdraw Wrapped Sonic (wS) from this strategy contract.
+    /// Used only if some wS is lingering on the contract.
+    /// That can happen when:
+    ///   - after mints if the strategy is the default
```

Comment:
Is this comment accurate?

Reply:
thanks it was not: https://github.com/OriginProtocol/origin-dollar/pull/2335/commits/11f967f74a90b828b5ae97a0a919cef3dfd2b72a